hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/ExchangeRateManager.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.workflows.handle;
+
+import static com.hedera.node.app.service.file.impl.FileServiceImpl.BLOBS_KEY;
+import static com.hederahashgraph.api.proto.java.ExchangeRateSet.parseFrom;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.hedera.hapi.node.base.FileID;
+import com.hedera.hapi.node.base.TimestampSeconds;
+import com.hedera.hapi.node.state.file.File;
+import com.hedera.hapi.node.transaction.ExchangeRate;
+import com.hedera.hapi.node.transaction.ExchangeRateSet;
+import com.hedera.node.app.service.file.FileService;
+import com.hedera.node.app.spi.state.ReadableKVState;
+import com.hedera.node.app.state.HederaState;
+import com.hedera.node.config.data.FilesConfig;
+import com.swirlds.config.api.Configuration;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@Singleton
+public class ExchangeRateManager {
+    private static final Logger log = LogManager.getLogger(ExchangeRateManager.class);
+
+    private int currHbarEquiv;
+    private int currCentEquiv;
+    private long currExpiry;
+    private int nextHbarEquiv;
+    private int nextCentEquiv;
+    private long nextExpiry;
+    private boolean isInitiated;
+    private ExchangeRateSet exchangeRateSet;
+
+    @Inject
+    public ExchangeRateManager() {
+        // For dagger
+    }
+
+    public void createUpdateExchangeRates(final HederaState hederaState, final Configuration configuration)
+            throws InvalidProtocolBufferException {
+        final var readableStates = hederaState.createReadableStates(FileService.NAME);
+        final ReadableKVState<FileID, File> files = readableStates.get(BLOBS_KEY);
+        final var fileConfig = configuration.getConfigData(FilesConfig.class);
+        final var fileId =
+                FileID.newBuilder().fileNum(fileConfig.exchangeRates()).build();
+        final var exchangeRateFile = files.get(fileId);
+
+        final var exchangeRateSetProto = parseFrom(exchangeRateFile.contents().toByteArray());
```

Comment:
The contents of this file are just normal encoded protobuf -- you can just parse with PBJ ExchangeRateSet Codec, and then you don't need to use the google protobuf objects, just PBJ, and you can skip the "convertRatesFromProto" method as well.

Reply:
Done, thanks for noticing.